searchsummary/src/vespa/searchsummary/docsummary/check_undefined_value_visitor.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "check_undefined_value_visitor.h"
+#include <vespa/searchcommon/common/undefinedvalues.h>
+#include <vespa/document/fieldvalue/fieldvalues.h>
+
+using search::attribute::isUndefined;
+
+
+namespace search::docsummary {
+
+void
+CheckUndefinedValueVisitor::visit(const document::AnnotationReferenceFieldValue&)
+{
+}
+
+void
+CheckUndefinedValueVisitor::visit(const document::ArrayFieldValue& value)
+{
+    if (value.isEmpty()) {
+        _is_undefined = true;
+    }
+}
+
+void
+CheckUndefinedValueVisitor::visit(const document::BoolFieldValue&)
+{
+}
+
+void
+CheckUndefinedValueVisitor::visit(const document::ByteFieldValue& value)
+{
+    if (isUndefined(value.getValue())) {
+        _is_undefined = true;
+    }
+}
+
+void
+CheckUndefinedValueVisitor::visit(const document::Document&)
+{
+}
+
+void
+CheckUndefinedValueVisitor::visit(const document::DoubleFieldValue& value)
+{
+    if (isUndefined(value.getValue())) {
```

Comment:
Why is this necessary ?
Single value fields are only undefined if they do not exist.
The special undefined value is only used in attributes, and should hence never make it into any document.

Reply:
This has been resolved in https://github.com/vespa-engine/vespa/pull/23317.